backend/src/app.controller.ts

```diff
@@ -9,4 +27,33 @@ export class AppController {
   getHello(): string {
     return this.appService.getHello();
   }
+
+  @Get('/files')
+  getAllFiles(): string {
+    return db.getData('/files');
+  }
+
+  @Get('/files/:id')
+  getFileById(@Param() params): string {
+    return db.getData(`/files/${params.id}`);
+  }
+
+  @Post('upload')
+  @UseInterceptors(FileInterceptor('file'))
+  async uploadFile(@UploadedFile() file: Express.Multer.File) {
+    const nftstorage = new NFTStorage({ token: NFT_STORAGE_KEY });
+    const results = await nftstorage.store({
+      image: new File([file.buffer], file.originalname),
+      name: file.originalname,
+      description: 'Bootcamp week4 - group1 - NFT collection',
+      ...{},
+    });
+    const dataToSave = {
+      cid: results.ipnft,
+      url: results.url,
+      resolvedUrl: `https://ipfs.io/ipfs/${results.ipnft}/metadata.json`,
```

Comment:
According to the [doc](https://docs.ipfs.io/how-to/address-ipfs-on-web/#dweb-addressing-in-brief) , the URL format looks like 
```
https://ipfs.io/ipfs/<CID>
# e.g
https://ipfs.io/ipfs/Qme7ss3ARVgxv6rXqVPiikMJ8u2NLgmgszg13pYrDKEoiu
```

Do we need the `metadata.json` ?

Reply:
@j-c17  yeah we need that to resolve  metadata for NFT. this is also mentioned in doc
<img width="748" alt="image" src="https://user-images.githubusercontent.com/7422971/179358110-f43d7e62-1d04-4448-98d2-7d32a85ebdd2.png">

here
https://docs.ipfs.io/how-to/best-practices-for-nft-data/#types-of-ipfs-links-and-when-to-use-them

so for example when u have NFT contract with URI storage
<img width="697" alt="image" src="https://user-images.githubusercontent.com/7422971/179358138-bc31cdcc-a1d5-4f2c-9496-071ba20314ea.png">

you can call method - 
```    
 function safeMint(address to, uint256 tokenId, string memory uri)
```

with params
```
safeMint("[address]",[tokeid], "ipfs://bafybeibnsoufr2renqzsh347nrx54wcubt5lgkeivez63xvivplfwhtpym/metadata.json")

```

since its a standard format/structure opensea and other marketplaces will be able to resolve ipfs uri (for minting we even dont need to pass https resolved urls, we can pass ipfs url like mentioned above. i've did this additional url with https for frontend just in case it will be needed)

Reply:
Awesome. Thanks @drsmog for sharing the information. 🙏